config/config.json

```diff
@@ -16,220 +16,27 @@
-        "misc-string-constructor" : ["default", "sensitive", "extreme"],
-        "misc-string-integer-assignment" : ["sensitive", "extreme"],
-        "misc-string-literal-with-embedded-nul" : ["default", "sensitive", "extreme"],
-        "misc-suspicious-enum-usage" : ["default", "sensitive", "extreme"],
-        "misc-suspicious-missing-comma" : ["default", "sensitive", "extreme"],
-        "misc-suspicious-semicolon" : ["default", "sensitive", "extreme"],
-        "misc-suspicious-string-compare" : ["sensitive", "extreme"],
-        "misc-swapped-arguments" : ["default", "sensitive", "extreme"],
-        "misc-unconventional-assign-operator" : ["default", "sensitive", "extreme"],
-        "misc-uniqueptr-reset-release" : ["default", "sensitive", "extreme"],
-        "misc-unused-alias-decls" : ["sensitive", "extreme"],
-        "misc-unused-parameters" : ["extreme"], 
-        "misc-unused-raii" : ["default", "sensitive", "extreme"],
-        "misc-unused-using-decls" : ["sensitive", "extreme"],
-        "misc-use-after-move" : ["default", "sensitive", "extreme"],
-        "misc-virtual-near-miss" : ["default", "sensitive", "extreme"],
-        "modernize-deprecated-headers" : ["sensitive", "extreme"],
-        "modernize-replace-auto-ptr" : ["sensitive", "extreme"],
-        "modernize-replace-random-shuffle" : ["sensitive", "extreme"],
-        "mpi-buffer-deref" : ["sensitive", "extreme"],
-        "mpi-type-mismatch" : ["sensitive", "extreme"],
-        "performance-faster-string-find" : ["sensitive", "extreme"],
-        "performance-for-range-copy" : ["sensitive", "extreme"],
-        "performance-implicit-conversion-in-loop" : ["sensitive", "extreme"],
-        "performance-inefficient-algorithm" : ["default", "sensitive", "extreme"],
-        "performance-inefficient-string-concatenation" : ["sensitive", "extreme"],
-        "performance-inefficient-vector-operation" : ["sensitive", "extreme"],
-        "performance-move-const-arg" : ["default", "sensitive", "extreme"],
-        "performance-noexcept-move-constructor" : ["default", "sensitive", "extreme"],
-        "performance-type-promotion-in-math-fn" : ["sensitive", "extreme"],
-        "performance-unnecessary-copy-initialization" : ["sensitive", "extreme"],
-        "performance-unnecessary-value-param" : ["sensitive", "extreme"]
-      }
-    }
+      "clangsa" : [{
+          "config_name": "default",
+          "version_regex": ".*",
+          "profiles":{
+              "default": "-e apiModeling -e core -e cplusplus -e deadcode -e nullability.NullPassedToNonnull -e nullability.NullReturnedFromNonnull -e optin.cplusplus -e optin.portability -e security.FloatLoopCounter -e security.insecureAPI.UncheckedReturn -e security.insecureAPI.getpw -e security.insecureAPI.gets -e security.insecureAPI.mkstemp -e security.insecureAPI.mktemp -e security.insecureAPI.rand -e security.insecureAPI.vfork -e unix -e valist",
```

Comment:
There is a lot of duplication of `-e` option. Can not we use the following form instead of this?
```json
default: {
    "enable": ["apiModeling", "core", "..." ],
    "disable": null
}
```

Reply:
The '-e' format is necessary because there is also the option to disable a checker explicitly. This is planned also in the future versions of checker configs, so this flexibility would be beneficial IMHO.

Reply:
I have updated the patch to only use enable lists (the same way it is done in bcae35339606ca758aa8ca9505c6956e0768e18a).